### PR TITLE
fix: move PRODUCTION bool to build time

### LINKED ISSUE #8627: chore: on server, Replace PROD with __PRODUCTION__
`PROD` is a remnant of the old days before we webpacked our server. Now that we run it through webpack first, we can use __PRODUCTION__ and that'll tree shake a few conditionals.

AC
- [] remove `PROD` and replace it with `__PRODUCTION__` on the server

### PR DESCRIPTION
**Pull request by mattkrick**
# Description

fix #8627

`__PRODUCTION__` is a webpack variable that is calculated at build time. 
Since this var is a constant at runtime, webpack can tree shake conditional branches & remove bits of unreachable code, e.g. `if (!__PRODUCTION__) doThing()` gets removed entirely & `doThing()` may also get removed if it is not used elsewhere.

## Demo

If CI passes it works!

### CODE REVIEW

**Comment by tghanken** [comment]
I'm going to remove myself as a reviewer, since this seems like a developer change 😂.

**Comment by mattkrick** [comment]
ah good catch, bad github bot

File: packages/server/graphql/CompiledQueryCache.ts [hunk 1]
```diff
@@ -45,7 +44,7 @@ export default class CompiledQueryCache {
     if (compiledQuery) return compiledQuery
     const r = await getRethink()
     let queryString = await r.table('QueryMap').get(docId)('query').default(null).run()
-    if (!queryString && !PROD) {
+    if (!queryString && !__PRODUCTION__) {
```
Comment (mattkrick) [inline packages/server/graphql/CompiledQueryCache.ts:47]: Verified working. Here's what it looks like when we run `yarn build`. No reference to `queryMap.json` in prod 🎉 

Screenshot 2023-09-06 at 3 27 15 PM

**Comment by mattkrick** [comment]
verified working when built!

**Pull request merged**